### PR TITLE
Bump botocore from 1.34.90 to 1.34.106

### PR DESCRIPTION
**Pull request by ferruzzi**
Follow-up to https://github.com/apache/airflow/pull/40052

I was able to get it a little higher.  After a bit more experimentation, this looks like the current practical version cap until aiobotocore does another release.

botocore changelog [here]

```

1.34.106
========

* api-change:``bedrock-agent-runtime``: Updating Bedrock Knowledge Base Metadata & Filters feature with two new filters listContains and stringContains
* api-change:``codebuild``: CodeBuild Reserved Capacity VPC Support
* api-change:``datasync``: Task executions now display a CANCELLING status when an execution is in the process of being cancelled.
* api-change:``grafana``: This release adds new ServiceAccount and ServiceAccountToken APIs.
* api-change:``medical-imaging``: Added support for importing medical imaging data from Amazon S3 buckets across accounts and regions.
* api-change:``securityhub``: Documentation-only update for AWS Security Hub


1.34.105
========

* api-change:``connect``: Amazon Connect provides enhanced search capabilities for flows & flow modules on the Connect admin website and programmatically using APIs. You can search for flows and flow modules by name, description, type, status, and tags, to filter and identify a specific flow in your Connect instances.
* api-change:``s3``: Updated a few x-id in the http uri traits


1.34.104
========

* api-change:``events``: Amazon EventBridge introduces KMS customer-managed key (CMK) encryption support for custom and partner events published on EventBridge Event Bus (including default bus) and UpdateEventBus API.
* api-change:``vpc-lattice``: This release adds TLS Passthrough support. It also increases max number of target group per rule to 10.


1.34.103
========

* api-change:``discovery``: add v2 smoke tests and smithy smokeTests trait for SDK testing
* api-change:``greengrassv2``: Mark ComponentVersion in ComponentDeploymentSpecification as required.
* api-change:``sagemaker``: Introduced support for G6 instance types on Sagemaker Notebook Instances and on SageMaker Studio for JupyterLab and CodeEditor applications.
* api-change:``sso-oidc``: Updated request parameters for PKCE support.


1.34.102
========

* api-change:``bedrock-agent-runtime``: This release adds support to provide guardrail configuration and modify inference parameters that are then used in RetrieveAndGenerate API in Agents for Amazon Bedrock.
* api-change:``pinpoint``: This release adds support for specifying email message headers for Email Templates, Campaigns, Journeys and Send Messages.
* api-change:``route53resolver``: Update the DNS Firewall settings to correct a spelling issue.
* api-change:``ssm-sap``: Added support for application-aware start/stop of SAP applications running on EC2 instances, with SSM for SAP
* api-change:``verifiedpermissions``: Adds policy effect and actions fields to Policy API's.


1.34.101
========

* api-change:``cognito-idp``: Add EXTERNAL_PROVIDER enum value to UserStatusType.
* api-change:``ec2``: Adding Precision Hardware Clock (PHC) to public API DescribeInstanceTypes
* api-change:``ecr``: This release adds pull through cache rules support for GitLab container registry in Amazon ECR.
* api-change:``fms``: The policy scope resource tag is always a string value, either a non-empty string or an empty string.
* api-change:``polly``: Add new engine - generative - that builds the most expressive conversational voices.
* api-change:``sqs``: This release adds MessageSystemAttributeNames to ReceiveMessageRequest to replace AttributeNames.


1.34.100
========

* api-change:``b2bi``: Documentation update to clarify the MappingTemplate definition.
* api-change:``budgets``: This release adds tag support for budgets and budget actions.
* api-change:``resiliencehub``: AWS Resilience Hub has expanded its drift detection capabilities by introducing a new type of drift detection - application resource drift. This new enhancement detects changes, such as the addition or deletion of resources within the application's input sources.
* api-change:``route53profiles``: Doc only update for Route 53 profiles that fixes some link  issues


1.34.99
=======

* api-change:``medialive``: AWS Elemental MediaLive now supports configuring how SCTE 35 passthrough triggers segment breaks in HLS and MediaPackage output groups. Previously, messages triggered breaks in all these output groups. The new option is to trigger segment breaks only in groups that have SCTE 35 passthrough enabled.


1.34.98
=======

* api-change:``bedrock-agent``: This release adds support for using Provisioned Throughput with Bedrock Agents.
* api-change:``connect``: This release adds 5 new APIs for managing attachments: StartAttachedFileUpload, CompleteAttachedFileUpload, GetAttachedFile, BatchGetAttachedFileMetadata, DeleteAttachedFile. These APIs can be used to programmatically upload and download attachments to Connect resources, like cases.
* api-change:``connectcases``: This feature supports the release of Files related items
* api-change:``datasync``: Updated guidance on using private or self-signed certificate authorities (CAs) with AWS DataSync object storage locations.
* api-change:``inspector2``: This release adds CSV format to GetCisScanReport for Inspector v2
* api-change:``sagemaker``: Amazon SageMaker Inference now supports m6i, c6i, r6i, m7i, c7i, r7i and g5 instance types for Batch Transform Jobs
* api-change:``sesv2``: Adds support for specifying replacement headers per BulkEmailEntry in SendBulkEmail in SESv2.


1.34.97
=======

* api-change:``dynamodb``: This release adds support to specify an optional, maximum OnDemandThroughput for DynamoDB tables and global secondary indexes in the CreateTable or UpdateTable APIs. You can also override the OnDemandThroughput settings by calling the ImportTable, RestoreFromPointInTime, or RestoreFromBackup APIs.
* api-change:``ec2``: This release includes a new API for retrieving the public endorsement key of the EC2 instance's Nitro Trusted Platform Module (NitroTPM).
* api-change:``personalize``: This releases ability to delete users and their data, including their metadata and interactions data, from a dataset group.
* api-change:``redshift-serverless``: Update Redshift Serverless List Scheduled Actions Output Response to include Namespace Name.


1.34.96
=======

* api-change:``bedrock-agent``: This release adds support for using MongoDB Atlas as a vector store when creating a knowledge base.
* api-change:``ec2``: Documentation updates for Amazon EC2.
* api-change:``personalize-runtime``: This release adds support for a Reason attribute for predicted items generated by User-Personalization-v2.
* api-change:``securityhub``: Updated CreateMembers API request with limits.
* api-change:``sesv2``: Fixes ListContacts and ListImportJobs APIs to use POST instead of GET.


1.34.95
=======

* api-change:``chime-sdk-voice``: Due to changes made by the Amazon Alexa service, GetSipMediaApplicationAlexaSkillConfiguration and PutSipMediaApplicationAlexaSkillConfiguration APIs are no longer available for use. For more information, refer to the Alexa Smart Properties page.
* api-change:``codeartifact``: Add support for the Ruby package format.
* api-change:``fms``: AWS Firewall Manager now supports the network firewall service stream exception policy feature for accounts within your organization.
* api-change:``omics``: Add support for workflow sharing and dynamic run storage
* api-change:``opensearch``: This release enables customers to create Route53 A and AAAA alias record types to point custom endpoint domain to OpenSearch domain's dualstack search endpoint.
* api-change:``pinpoint-sms-voice-v2``: Amazon Pinpoint has added two new features Multimedia services (MMS) and protect configurations. Use the three new MMS APIs to send media messages to a mobile phone which includes image, audio, text, or video files. Use the ten new protect configurations APIs to block messages to specific countries.
* api-change:``qbusiness``: This is a general availability (GA) release of Amazon Q Business. Q Business enables employees in an enterprise to get comprehensive answers to complex questions and take actions through a unified, intuitive web-based chat experience - using an enterprise's existing content, data, and systems.
* api-change:``quicksight``: New Q embedding supporting Generative Q&A
* api-change:``route53resolver``: Release of FirewallDomainRedirectionAction parameter on the Route 53 DNS Firewall Rule.  This allows customers to configure a DNS Firewall rule to inspect all the domains in the DNS redirection chain (default) , such as CNAME, ALIAS, DNAME, etc., or just the first domain and trust the rest.
* api-change:``sagemaker``: Amazon SageMaker Training now supports the use of attribute-based access control (ABAC) roles for training job execution roles. Amazon SageMaker Inference now supports G6 instance types.
* api-change:``signer``: Documentation updates for AWS Signer. Adds cross-account signing constraint and definitions for cross-account actions.


1.34.94
=======

* api-change:``amplify``: Updating max results limit for listing any resources (Job, Artifacts, Branch, BackendResources, DomainAssociation) to 50 with the exception of list apps that where max results can be up to 100.
* api-change:``connectcases``: This feature releases DeleteField, DeletedLayout, and DeleteTemplate API's
* api-change:``inspector2``: Update Inspector2 to include new Agentless API parameters.
* api-change:``timestream-query``: This change allows users to update and describe account settings associated with their accounts.
* api-change:``transcribe``: This update provides error messaging for generative call summarization in Transcribe Call Analytics
* api-change:``trustedadvisor``: This release adds the BatchUpdateRecommendationResourceExclusion API to support batch updates of Recommendation Resource exclusion statuses and introduces a new exclusion status filter to the ListRecommendationResources and ListOrganizationRecommendationResources APIs.


1.34.93
=======

* api-change:``codepipeline``: Add ability to manually and automatically roll back a pipeline stage to a previously successful execution.
* api-change:``cognito-idp``: Add LimitExceededException to SignUp errors
* api-change:``connectcampaigns``: This release adds support for specifying if Answering Machine should wait for prompt sound.
* api-change:``marketplace-entitlement``: Releasing minor endpoint updates.
* api-change:``oam``: This release introduces support for Source Accounts to define which Metrics and Logs to share with the Monitoring Account
* api-change:``rds``: SupportsLimitlessDatabase field added to describe-db-engine-versions to indicate whether the DB engine version supports Aurora Limitless Database.
* api-change:``support``: Releasing minor endpoint updates.


1.34.92
=======

* api-change:``appsync``: UpdateGraphQLAPI documentation update and datasource introspection secret arn update
* api-change:``fms``: AWS Firewall Manager adds support for network ACL policies to manage Amazon Virtual Private Cloud (VPC) network access control lists (ACLs) for accounts in your organization.
* api-change:``ivs``: Bug Fix: IVS does not support arns with the `svs` prefix
* api-change:``ivs-realtime``: Bug Fix: IVS Real Time does not support ARNs using the `svs` prefix.
* api-change:``rds``: Updates Amazon RDS documentation for setting local time zones for RDS for Db2 DB instances.
* api-change:``stepfunctions``: Add new ValidateStateMachineDefinition operation, which performs syntax checking on the definition of a Amazon States Language (ASL) state machine.


1.34.91
=======

* api-change:``datasync``: This change allows users to disable and enable the schedules associated with their tasks.
* api-change:``ec2``: Launching capability for customers to enable or disable automatic assignment of public IPv4 addresses to their network interface
* api-change:``emr-containers``: EMRonEKS Service support for SecurityConfiguration enforcement for Spark Jobs.
* api-change:``entityresolution``: Support Batch Unique IDs Deletion.
* api-change:``gamelift``: Amazon GameLift releases container fleets support for public preview. Deploy Linux-based containerized game server software for hosting on Amazon GameLift.
* api-change:``ssm``: Add SSM DescribeInstanceProperties API to public AWS SDK.

```

### CODE REVIEW

File: airflow/providers/amazon/provider.yaml [hunk 1]
```diff
@@ -135,9 +135,9 @@ devel-dependencies:
   - aws_xray_sdk>=2.12.0
   - moto[cloudformation,glue]>=5.0.0
   - mypy-boto3-appflow>=1.34.0
-  - mypy-boto3-rds>=1.34.90
+  - mypy-boto3-rds>=1.34.106
   - mypy-boto3-redshift-data>=1.34.0
-  - mypy-boto3-s3>=1.34.90
+  - mypy-boto3-s3>=1.34.105
```
Comment (ferruzzi) [inline airflow/providers/amazon/provider.yaml:140]: Not a typo, 105 is the latest release on this one.

**Review by vincbeck — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Comment by jedcunningham** [comment]
For my own edification, is there a reason we need to bump the min version?

**Comment by ferruzzi** [comment]
@jedcunningham   We needed 1.34.90 for a feature that was requested, and I did that bump yesterday.  So no, the bump from x.90 to x.106 isn't currently mandatory, but there are a lot of API changes there to use (or not).      

Do you think we should leave this at x.90 until we NEED it?

**Comment by jedcunningham** [comment]
Generally speaking, yes, we should set the min based on what we actually need from that dep. Of course, constraints can and will have the higher version.

**Comment by jedcunningham** [comment]
That's why the `--resolution lowest` flag on uv is appealing btw, it'll let us identify when we've missed bumping the min on a dep :) (though I'm not sure if we are doing that yet or not, can't remember as I haven't followed that closely).

**Comment by potiuk** [comment]
> That's why the --resolution lowest flag on uv is appealing btw, it'll let us identify when we've missed bumping the min on a dep :) (though I'm not sure if we are doing that yet or not, can't remember as I haven't followed that closely).

 Precisely. We should not **just** bump the dependency with no reason. There could be good reasons which is something we know (like avoiding a security issue) but generally the `--lowest-direct` test should find and point out the cases where we need to do the bump because some of our code (and we know it via unit test) needs a feature or fix added in that version.

**Comment by potiuk** [comment]
And yes we are doing it alread - merged 2 days ago :)

**Review by potiuk — CHANGES_REQUESTED** [review]
So ... requesting changes ... in case there is no **real** need to do the bump we should not do it.

**Comment by potiuk** [comment]
BTW. For botocore specifically, we might have another reason to bump it - due to botocore releasing almost every day, bumping it from time to time to just limit the available number of candidates for resolution algorithm is generally a good idea - becasuse some of the heuristics there might get it into long backtracking - but we will also see that generally in CI and that's when we have ~ 80 or 100 candidates we should worry about it. It's not the case now, and generally when we continue adding operators and features to the provider with `--lowest-direct` and tests detecting need for a bump, the "we need it" bump should happen frequently enough to no worry about this case any more.

**Pull request closed without merging**